carbon-mb/components/transports/mqtt/src/main/java/org/wso2/carbon/andes/transports/mqtt/adaptors/andes/message/MqttMessageContext.java

```diff
@@ -134,4 +217,71 @@ public boolean isCompressed() {
         return isCompressed;
     }
 
+    /**
+     * Get the {@link AndesMessage} representation of {@link MqttMessageContext}
+     * @return AndesMessage
+     */
+    public AndesMessage toAndesMessage() {
+
+        AndesMessageMetadata andesMetadata = new AndesMessageMetadata(getTopic(), protocolType);
+        andesMetadata.setTopic(true);
+        andesMetadata.setPersistent(true);
+        andesMetadata.setRetain(isRetain());
```

Comment:
is retain a common property or specific to MQTT? If not common it should go to properties.